Add a little tiny interesting word exam

EdgeEden opened the pull request.

EdgeEden committed:
Add files via upload

EdgeEden committed:
Add files via upload

EdgeEden committed:
Add files via upload

faf4r commented on bot.py:
```diff
@@ -236,6 +263,6 @@ async def on_message_create(self, message: GroupMessage):
 
 
-intents = botpy.Intents(public_messages=True, guild_messages=True)
+intents = botpy.Intents(public_messages=True, guild_messages=False)
```
这个改不改没什么影响的吧，guild_message是频道的，虽然因为指令过审那边就没配置频道了

EdgeEden commented:
没有影响 但是我不这么改的话没过审的机器人用不了捏



---Original---
From: ***@***.***&gt;
Date: Thu, Apr 3, 2025 20:31 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [faf4r/QQbot-botpy] Add a little tiny interesting word exam (PR#1)




 
@faf4r commented on this pull request.
 
 
In bot.py:
 &gt; @@ -236,6 +263,6 @@ async def on_message_create(self, message: GroupMessage):          logger.debug(msg_result)     -intents = botpy.Intents(public_messages=True, guild_messages=True) +intents = botpy.Intents(public_messages=True, guild_messages=False)  
这个改不改没什么影响的吧，guild_message是频道的，虽然因为指令过审那边就没配置频道了
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

faf4r commented:
我记得我没过审的时候也是能用的啊，哦，因为我是私域机器人，你是公域是吧

faf4r reviewed (changes requested):
1. get_word_with_definition可以不要num参数，每次exam一个单词就挺好。
2. get_possible_answers返回set就可以了，不用转成list，且add进去的时候就可以lower单词了，后面直接判断in就行了
3. 回答问题也用一个指令标明，便于处理也不影响其它功能
4. bot.py的97行，如果用户传入的tag没有内容的话要直接处理了，后面没必要查询possible_answers和加入pending_word
5. TODO：后续可以考虑用词向量来个给英文回答中文的，不过我得研究一下先

EdgeEden commented:
有空就改😣
怎么有一种甲方的感觉😇



---Original---
From: ***@***.***&gt;
Date: Fri, Apr 4, 2025 10:53 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [faf4r/QQbot-botpy] Add a little tiny interesting word exam (PR#1)




 
@faf4r requested changes on this pull request.
  
get_word_with_definition可以不要num参数，每次exam一个单词就挺好。
 
get_possible_answers返回set就可以了，不用转成list，且add进去的时候就可以lower单词了，后面直接判断in就行了
 
回答问题也用一个指令标明，便于处理也不影响其它功能
 
bot.py的97行，如果用户传入的tag没有内容的话要直接处理了，后面没必要查询possible_answers和加入pending_word
 
TODO：后续可以考虑用词向量来个给英文回答中文的，不过我得研究一下先
  
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

EdgeEden committed:
Merge branch 'faf4r:main' into main

EdgeEden committed:
Add files via upload

EdgeEden committed:
Modified two funtions

EdgeEden committed:
Modified little word exam

Add command /answer to answer the exam as requested
And a little more robust improved as requested

faf4r commented on bot.py:
```diff
@@ -82,6 +87,34 @@ async def handle_msg(self, message: GroupMessage):
+            logger.info(content)
+            return await message.reply(content=content)
+
+        elif msg.lower().split()[0] in ["/answer", "answer", "/回答", "回答"] and user_id in self.pending_word[group_id]:
```
这里要如果没有pending word的话要给出提示，不然会触发最下面的else开AI对话了

EdgeEden committed:
fix(WordExam):Modified answer receive function

EdgeEden committed:
Update README.md

faf4r committed:
Merge branch 'main' into main
